### PR TITLE
Add meta information about the currency

### PR DESCRIPTION
**Pull request by mitchelli**

### CODE REVIEW

**Review by thepiwo — APPROVED** [review]
I don't think there has been a specific wallet integration discussed yet, but the information provided should be sufficient.

File: apps/aeutils/priv/aeternity_config_schema.json [hunk 1]
```diff
@@ -1562,6 +1562,61 @@
...
+                "display" : {
+                    "type" : "object",
+                    "additionalProperties" : false,
+                    "properties" : {
+                        "logo_file": {
+                            "description": "Logo file of the chain.",
+                            "type": "string"
+                        },
+                        "primary_colour": {
+                            "description": "Short name of currency.",
```
Comment (dincho) [inline apps/aeutils/priv/aeternity_config_schema.json:1604]: wrong description :)

File: apps/aeutils/priv/aeternity_config_schema.json [hunk 2]
```diff
@@ -1562,6 +1562,61 @@
...
+                            "type": "string",
+                            "default": "#f5274e"
+                        },
+                        "secondary_colour" : {
+                            "description" : "Smallest denomination of currency.",
```
Comment (dincho) [inline apps/aeutils/priv/aeternity_config_schema.json:1609]: wrong description

File: apps/aeutils/priv/aeternity_config_schema.json [hunk 3]
```diff
@@ -1562,6 +1562,61 @@
...
+                            "type" : "string",
+                            "default" : "#242424"
+                        },
+                        "network_id": {
```
Comment (dincho) [inline apps/aeutils/priv/aeternity_config_schema.json:1613]: Shouldn't it be `network_name`, we already have the network_id in the status endpoint?
And the corresponding default "Mainnet"

File: apps/aeutils/priv/aeternity_config_schema.json [hunk 4]
```diff
@@ -1562,6 +1562,61 @@
                         }
                     }
                 },
+                "currency" : {
+                    "type" : "object",
+                    "additionalProperties" : false,
+                    "properties" : {
+                        "long_name": {
```
Comment (dincho) [inline apps/aeutils/priv/aeternity_config_schema.json:1569]: perhaps `long_name` should be just `name` and `short_name` should be symbol 🤔  just a suggestion 💭 

I'm not sure if the properties schema has been agreed somewhere else already or this is first proposition

File: apps/aeutils/priv/aeternity_config_schema.json [hunk 5]
```diff
@@ -1562,6 +1562,61 @@
...
+                            "description": "Long name of currency.",
+                            "type": "string",
+                            "default": "aeternity"
+                        },
+                        "short_name": {
+                            "description": "Short name of currency.",
+                            "type": "string",
+                            "default": "AE"
+                        },
+                        "smallest_denomination" : {
```
Comment (dincho) [inline apps/aeutils/priv/aeternity_config_schema.json:1579]: Alternative names would be `subunit` or `minor_units`

- https://en.wikipedia.org/wiki/Denomination_(currency)
- https://english.stackexchange.com/questions/223121/is-there-a-word-for-the-most-granular-parts-of-currency-eg-cents-pence-etc

**Review by dincho — COMMENTED** [review]
just naming comments

File: apps/aeutils/priv/aeternity_config_schema.json (hunk 3 above)
Comment (mitchelli) [reply apps/aeutils/priv/aeternity_config_schema.json:1613]: Yes, I think this makes more sense, this is just for display purposes.

**Comment by hanssv** [comment]
I don't know the background - should there be defaults for Mainnet and Testnet perhaps?

**Comment by mitchelli** [comment]
> I don't know the background - should there be defaults for Mainnet and Testnet perhaps?

The purpose of this PR was to allow the wallet to get information about a hyper chain but it should also work with the regular chain. I could change the default for network_name to network_id if network_name is not set, rather than "mainnet"?

**Comment by dincho** [comment]
> I don't know the background - should there be defaults for Mainnet and Testnet perhaps?

good point

**Review by hanssv — APPROVED** [review]

**Review by dincho — APPROVED** [review]

**Pull request merged**